### ISSUE TITLE
Provider Configuration System

### ISSUE THREAD

**Issue by manavgup** [manavgup/rag_modulo#89]
## Objective
Implement a provider configuration and registry system that allows dynamic registration and management of LLM providers.

## Scope
- [ ] Create new files:
  - `models/provider_config.py`
  - `schemas/provider_config_schema.py`
  - `repository/provider_config_repo.py`
  - `services/provider_config_service.py`

### 1. Models and Schemas
```python
class ProviderModelConfig(BaseModel):
    model_id: str
    provider_name: str
    is_active: bool = True
    last_verified: Optional[datetime]
    parameters: LLMParameters

class ProviderRegistry(BaseModel):
    providers: Dict[str, ProviderModelConfig] = {}

    def register_model(self, provider: str, model_id: str, config: LLMParameters):
        key = f"{provider}/{model_id}"
        self.providers[key] = ProviderModelConfig(
            model_id=model_id,
            provider_name=provider,
            parameters=config
        )
```

### 2. Repository Layer
- [ ] Implement BaseProviderRepository
- [ ] Add CRUD operations
- [ ] Add provider registration logic
- [ ] Add validation handling

### 3. Service Layer
- [ ] Implement ProviderConfigService
- [ ] Add provider management logic
- [ ] Add model registration
- [ ] Add validation rules
- [ ] Add error handling

### 4. Provider Management Features
- [ ] Dynamic provider registration
- [ ] Model configuration management
- [ ] Validation rules per provider
- [ ] Status tracking
- [ ] Version management

## Dependencies
- Core Configuration Cleanup (#85) should be completed first
- Should be implemented after LLM Parameters (#86)
- Can be parallel with Runtime Config (#88)

## Testing Requirements
- [ ] Provider registration tests
- [ ] Model configuration tests
- [ ] Validation tests
- [ ] CRUD operation tests
- [ ] Error handling tests

## Migration Steps
- [ ] Create database schema
- [ ] Migrate existing provider configs
- [ ] Add validation for existing configs
- [ ] Provide rollback capability

## PR Checklist
- [ ] Implement models and schemas
- [ ] Add repository layer
- [ ] Add service layer
- [ ] Add model registry
- [ ] Add comprehensive tests
- [ ] Update documentation
- [ ] Add database migrations
- [ ] Add migration utilities

**Comment by manavgup** [comment]
Closing as obsolete. This configuration system from Dec 2024 has been superseded by the current runtime configuration implementation. The provider configuration is now handled through the existing services/provider_service.py and repository/provider_repository.py with proper database-backed configuration.
